.travis.yml

```diff
@@ -7,13 +7,17 @@ git:
 before_install:
   - sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
   - sudo apt-get update -qq
-  - sudo apt-get install -y python-ply python-git libelf-dev codespell sparse fakeroot gcc-9 g++-9 gcc-aarch64-linux-gnu
+  - sudo apt-get install -y python-ply python-git libelf-dev codespell fakeroot gcc-9 g++-9 gcc-aarch64-linux-gnu
   - sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 90 --slave /usr/bin/g++ g++ /usr/bin/g++-9
   - sudo apt-get install -y xz-utils lftp
   - git clone https://github.com/thesofproject/kconfig.git
   - mkdir 0day
   - wget https://raw.githubusercontent.com/intel/lkp-tests/master/sbin/make.cross -O make.cross
   - chmod +x make.cross
+  - mkdir -p $PWD/bin
+  - git clone git://git.kernel.org/pub/scm/devel/sparse/sparse.git
```

Comment:
Can we stick to some tag to avoid non-determinism? If not, at the very least run `git describe` to have a record in the logs

Reply:
> A different commit for removing coccicheck would be nice and very easy and quick to test/review/approve.

Meh. It never worked, might as well clean everything at once.
 
> > Not sure what happened but Travis CI does not seem to start. Not great for a Travis CI update...
> 
> Whenever this happens please "save" the corresponding link because every (force) push loses it. I mean like this for instance:
> 
> https://travis-ci.org/github/thesofproject/linux/builds/710187112

Yes, but when Travis doesn't start, there's no link...
 
> Please make most of this a new `get-sparse.sh` script checked in one of the git repos so it can be run locally, see example at the bottom of:
> https://docs.travis-ci.com/user/installing-dependencies/#installing-projects-from-source

no, I don't want this in the kernel. we can do this in .travis.yml alone.
 
> Maybe because changing PATH and CWD in `before_install:` may not affect later phases? Error handling and reporting differences between phases tend to be subtle; better stick to (examples in) the official documentation. Error handling is of course never tested, neither in Travis nor anywhere else :-)

using the home directory failed so I used $PWD. it seems to do the job.

Reply:
> using the home directory failed

You were using `~` which behaves quite differently from `$HOME`, example: `echo "~"`. Travis has special shell tricks to run commands.